portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -45,31 +46,18 @@
 @MultipartConfig 
 public class CommentServlet extends HttpServlet {
 
+
+  private static final String TOPIC_NAME = "comment-updates";
+  private static final PublishHandler PUBLISH_HANDLER = new PublishHandler(TOPIC_NAME);
```

Comment:
By making this static, you've made it a singleton, and introduced a race condition if the same server gets two different requests and one overwrites PublishHandler's state (e.g. the publisher variable) while the other is still working with it.

In general, you almost never see singletons like this, or if you do they are explicitly marked as such and use the singleton design pattern (and the @Singleton decorator).  It is a big concurrency+scalability no-no.